Makefile

```diff
@@ -82,15 +68,70 @@ go_webdriver_test: go_webdriver_deps bower_components
 			--firefox_path=$(FIREFOX_PATH) \
 			--geckodriver_path=$(GECKODRIVER_PATH)
 
+sys_update: sys_deps
+	sudo apt-get update
+	gcloud components update
+	npm install -g npm
+
 go_webdriver_deps: go_deps webdriver_deps
 
 webdriver_deps:
+	sudo apt-get install --assume-yes --no-install-suggests default-jdk wget xvfb $$(apt-cache depends firefox-esr chromedriver |  grep Depends | sed "s/.*ends:\ //" | tr '\n' ' ')
+	cd $(WPTD_PATH)webapp; npm install web-component-tester --unsafe-perm
 	cd $(WEBDRIVER_PATH); ./install.sh $(BROWSERS_PATH)
 
-go_deps: $(find .  -type f | grep '\.go$' | grep -v '\.pb.go$')
+go_deps: sys_deps $(find .  -type f | grep '\.go$' | grep -v '\.pb.go$')
+	# Manual git clone + install is a workaround for #85.
+	if [ "$$(which golint)" == "" ]; \
+		then \
+		mkdir -p "$(GOPATH)/src/golang.org/x"; \
+		cd "$(GOPATH)/src/golang.org/x" && git clone https://github.com/golang/lint; \
+		cd "$(GOPATH)/src/golang.org/x/lint" && go get ./... && go install ./...; \
+	fi
 	cd $(WPTD_GO_PATH); go get -t -tags="small medium large" ./...
 
+sys_deps:
+	if [[ "$$(which curl)" == "" ]]; \
+		then \
+		sudo apt-get install --assume-yes --no-install-suggests curl; \
+	fi
+	if [[ "$$(which git)" == "" ]]; \
+		then \
+		sudo apt-get install --assume-yes --no-install-suggests git; \
+	fi
+	if [[ "$$(which python)" == "" ]]; \
+		then \
+		sudo apt-get install --assume-yes --no-install-suggests python; \
+	fi
+	if [[ "$$(which gpg)" == "" ]]; \
+	then \
+		sudo apt-get install --assume-yes --no-install-suggests gnupg; \
+	fi
+	if [[ "$$(which gcloud)" == "" ]]; \
+		then \
+		curl -s https://sdk.cloud.google.com > ./install-gcloud.sh; \
+		bash ./install-gcloud.sh --disable-prompts --install-dir=$(HOME); \
+		rm -f ./install-gcloud.sh; \
+		gcloud components install --quiet \
+			app-engine-go \
+			core \
+			gsutil \
+			app-engine-python; \
+		gcloud config set disable_usage_reporting false; \
+	fi
+	if [[ "$$(which nodejs)" == "" ]]; \
+	then \
+		curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -; \
+		sudo apt-get install --assume-yes --no-install-suggests nodejs; \
+	fi
+	if [[ "$$(which npm)" == "" ]]; \
```

Comment:
The `nodejs` package from `deb.nodesource.com` [already provides `npm`](https://deb.nodesource.com/node_8.x/dists/sid/main/binary-amd64/Packages), so there's no need to detect and install `npm` seperately (in fact, `apt-get install npm` will try to install `nodejs` again and hence be a no-op).